go/libkb/httpsrv.go

```diff
@@ -0,0 +1,60 @@
+package libkb
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+)
+
+// RandomPortHTTPSrv starts a simple HTTP server on a random port and and exposes all the methods
+// of http.ServeMux
+type RandomPortHTTPSrv struct {
+	sync.Mutex
+	server *http.Server
+	*http.ServeMux
+}
+
+func NewRandomPortHTTPSrv() *RandomPortHTTPSrv {
+	return &RandomPortHTTPSrv{}
+}
+
+func (h *RandomPortHTTPSrv) Start() (address string, err error) {
+	h.Lock()
+	defer h.Unlock()
+
+	h.ServeMux = http.NewServeMux()
+	minPort := 7000
+	maxPort := 16000
+	for p := minPort; p < maxPort; p++ {
```

Comment:
Can you use [this approach](https://stackoverflow.com/questions/43424787/how-to-use-next-available-port-in-http-listenandserve?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa) to have the OS assigned you a port?